lib/ContainerManager.js

```diff
@@ -408,6 +435,76 @@ Server.prototype.routes.get['containers'] = function(req, res, next) {
   });
 };
 
+/**
+ * Delete the docker container from the server. THIS OPERATION CANNOT BE REVERSED!
+ * By default, will not remove running containers.
+ * Call with ?force=true delete running containers too.
+ */
+Server.prototype.routes.del['containers/:id'] = function(req, res, next) {
+  var self = this
+
+  // delete/remove the container and update the build entry in the DB
+  var container = new Container({
+    docker: this.config.docker,
+    containerId: req.params.id,
+    log: this.log,
+  })
+
+  log_container_events(container)
+
+  function handle_error(err, msg){
+    msg = msg ? msg + ': ' : ''
+    var message = msg + (err.json || err.message)
+    var code = err.statusCode || 500
+    res.send(code, {error: message.trim(), code: code})
+    next()
+  }
+
+  container.inspect(function(err, info){
+    if(err){
+      return handle_error(err)
+    }
+
+    // name is: probo--[repo slug]--[project id]--[buildid]
+    var buildId = info.Name.split('--')[3]
+
+    if(!buildId){
+      var err = new Error(`BuildId not found for container name ${info.Name}`)
+      err.statusCode = 400
+      return handle_error(err)
+    }
+
+    container.remove({force: req.query.force, v: req.query.v}, function(err, data){
+      if(err) return handle_error(err)
+
+      // update the build object
+      self.getBuildData(buildId, function(err, build){
```

Comment:
This is basically composing 3 functions and some error handling. Lots of code, I know, but I'm not sure this combination would be useful anywhere else.